src/test/groovy/org/prebid/server/functional/tests/module/pbrequestcorrection/PbRequestCorrectionSpec.groovy

```diff
@@ -0,0 +1,449 @@
+    }
+
+    def "PBS shouldn't remove positive instl from imps for app when request correction is enabled for account but some required parameter is empty"() {
+        given: "Android APP bid request with version lover then version threshold"
+        def prebid = new AppPrebid(source: source, version: version)
+        def bidRequest = BidRequest.getDefaultBidRequest(APP).tap {
+            imp.first.instl = instl
+            app.bundle = bundle
+            app.ext = new AppExt(prebid: prebid)
+        }
+
+        and: "Account in the DB"
+        def requestCorrectionConfig = PbRequestCorrectionConfig.defaultConfigWithInterstitial
+        def account = createAccountWithRequestCorrectionConfig(bidRequest, requestCorrectionConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        pbsServiceWithRequestCorrectionModule.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain original imp.instl"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.imp.instl == bidRequest.imp.instl
+
+        where:
+        bundle  | source        | version                                       | instl
+        null    | PREBID_MOBILE | ACCEPTABLE_DEVICE_INSTL_PREBID_MOBILE_VERSION | YES
+        ANDROID | null          | ACCEPTABLE_DEVICE_INSTL_PREBID_MOBILE_VERSION | YES
+        ANDROID | PREBID_MOBILE | null                                          | YES
+        ANDROID | PREBID_MOBILE | ACCEPTABLE_DEVICE_INSTL_PREBID_MOBILE_VERSION | null
+    }
+
+    def "PBS shouldn't remove positive instl from imps for android app when request correction is enabled for account and version is 2.2.3"() {
+        given: "Android APP bid request with version equal to 2.2.3"
+        def prebid = new AppPrebid(source: PBSUtils.getRandomCase(PREBID_MOBILE), version: "2.2.3")
+        def bidRequest = BidRequest.getDefaultBidRequest(APP).tap {
+            imp.first.instl = YES
+            app.bundle = PBSUtils.getRandomCase(ANDROID)
+            app.ext = new AppExt(prebid: prebid)
+        }
+
+        and: "Account in the DB"
+        def requestCorrectionConfig = PbRequestCorrectionConfig.defaultConfigWithInterstitial
+        def account = createAccountWithRequestCorrectionConfig(bidRequest, requestCorrectionConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        pbsServiceWithRequestCorrectionModule.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain original imp.instl"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.imp.instl == bidRequest.imp.instl
+    }
+
+    def "PBS shouldn't remove positive instl from imps for android app when request correction is enabled for account and version is higher then 2.2.3"() {
+        given: "Android APP bid request with version higher then 2.2.3"
```

Comment:
given: "Android APP bid request with version higher then version threshold"
same for others